doc/api/net.md

```diff
@@ -743,84 +790,86 @@ buffer. Returns `false` if all or part of the data was queued in user memory.
 The optional `callback` parameter will be executed when the data is finally
 written out - this may not be immediately.
 
-## net.connect(options[, connectListener])
-<!-- YAML
-added: v0.7.0
--->
-
-A factory function, which returns a new [`net.Socket`][] and automatically
-connects with the supplied `options`.
-
-The options are passed to both the [`net.Socket`][] constructor and the
-[`socket.connect`][] method.
+## net.connect()
 
-The `connectListener` parameter will be added as a listener for the
-[`'connect'`][] event once.
+Aliases to
+[`net.createConnection()`][`net.createConnection()`].
 
-Here is an example of a client of the previously described echo server:
-
-```js
-const net = require('net');
-const client = net.connect({port: 8124}, () => {
-  // 'connect' listener
-  console.log('connected to server!');
-  client.write('world!\r\n');
-});
-client.on('data', (data) => {
-  console.log(data.toString());
-  client.end();
-});
-client.on('end', () => {
-  console.log('disconnected from server');
-});
-```
+Possible signatures:
 
-To connect on the socket `/tmp/echo.sock` the second line would just be
-changed to
+* [`net.connect(options[, connectListener])`][`net.connect(options)`]
+* [`net.connect(path[, connectListener])`][`net.connect(path)`] for connections
+  to UNIX domain sockets or Windows named pipes.
+* [`net.connect(port[, host][, connectListener])`][`net.connect(port, host)`]
+  for TCP connections.
 
-```js
-const client = net.connect({path: '/tmp/echo.sock'});
-```
+### net.connect(options[, connectListener])
+<!-- YAML
+added: v0.7.0
+-->
+Alias to
+[`net.createConnection(options[, connectListener])`][`net.createConnection(options)`].
 
-## net.connect(path[, connectListener])
+### net.connect(path[, connectListener])
 <!-- YAML
 added: v0.1.90
 -->
 
-A factory function, which returns a new unix [`net.Socket`][] and automatically
-connects to the supplied `path`.
-
-The `connectListener` parameter will be added as a listener for the
-[`'connect'`][] event once.
+Alias to
+[`net.createConnection(path[, connectListener])`][`net.createConnection(path)`].
 
-## net.connect(port[, host][, connectListener])
+### net.connect(port[, host][, connectListener])
 <!-- YAML
 added: v0.1.90
 -->
 
-A factory function, which returns a new [`net.Socket`][] and automatically
-connects to the supplied `port` and `host`.
+Alias to
+[`net.createConnection(port[, host][, connectListener])`][`net.createConnection(port, host)`].
 
-If `host` is omitted, `'localhost'` will be assumed.
+## net.createConnection()
 
-The `connectListener` parameter will be added as a listener for the
-[`'connect'`][] event once.
+A factory function, which creates a new [`net.Socket`][],
+immediately initiates connection with [`socket.connect()`][],
+then returns the `net.Socket` that starts the connection.
 
-## net.createConnection(options[, connectListener])
+This function is asynchronous. When the connection is established, a
```

Comment:
> This function is asynchronous.

I think this is misleading. The connection object is returned synchronously, but the establishment of the connection is asynchronous. Better would be something like:

> Creates a socket object and initiates a connection. When the connection is established....